BlueBrain/CoreNeuron
# NVHPC@21.11+ workaround for Random123.

**Description**
Mainly add a workaround for an issue in Random123 compiler detection logic with versions 21.11 and newer of the NVHPC compilers.

- Add `-DR123_USE_INTRIN_H=0` in NVHPC builds with GPU support disabled to avoid #724, as recommended in https://github.com/DEShawResearch/random123/issues/6.
- If GPU support is enabled we compile with `-cuda`, which means Random123 treats `nvc++` as `nvcc` and we do not see the issue.
- Fix some compiler warnings when compiling with NVHPC and GPU support disabled.

Closes https://github.com/BlueBrain/CoreNeuron/issues/724.
See also: https://github.com/neuronsimulator/nrn/pull/1587.

**How to test this?**
Build with NVHPC 21.11 or newer and GPU support disabled. Without this PR, you should see compilation errors.

**Use certain branches for the SimulationStack CI**

CI_BRANCHES:NEURON_BRANCH=master,
